### PR TITLE
Printing Tribonacci series upt to "n" number of terms

### LINKED ISSUE #3028: Printing Tribonacci series up to "n" number of terms.
## 💥 Proposal

This Algorithm prints Tribonacci series up to "n" terms where the value of "n" is to be provided by the user.
a(n)=a(n-1)+a(n-2)+a(n-3)

For Example:
(input) n=10
(output):  0, 0, 1, 1, 2, 4, 7, 13, 24, 44

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes.

### PR DESCRIPTION
**Pull request by AkanshaUniyal**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Prints the Tribonacci series till n terms.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixes #3028

### CODE REVIEW

File: Java/math/tribonacci.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+ package mypack;
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top

File: Java/math/tribonacci.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+ package mypack;
+ import java.util.Scanner;
+ public class tribonacci 
+ {
+    public static void main(String [] args)
+    {
+	/*Approach : A term is the sum of preceding three terms in this series. The sum of three preceding variables will be stored in "sum" variable.
+	  The first of the three variable terms will be overrided by second variable term which in turn will be overrided by third one and the third term will be 
+	  overrided by the sum. This process will continue till n terms.*/
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Shit your approach after description

File: Java/math/tribonacci.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ public class tribonacci 
+ {
+    public static void main(String [] args)
+    {
+	/*Approach : A term is the sum of preceding three terms in this series. The sum of three preceding variables will be stored in "sum" variable.
+	  The first of the three variable terms will be overrided by second variable term which in turn will be overrided by third one and the third term will be 
+	  overrided by the sum. This process will continue till n terms.*/ 
+
+        Scanner scan = new Scanner(System.in);
+        long num;int first=0,second=0,third=1,sum=0,count=0;
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Shift `int first=0,second=0,third=1,sum=0,count=0;` to new line

File: Java/math/tribonacci.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            System.out.print(", "+0);
+           else if(count==3)
+            System.out.print(", "+1);
+            else
+            {
+            sum=first+second+third;
+            System.out.print(", "+sum);
+            first=second;second=third;third=sum;
+            }
+         num--;
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: - Align your code properly
- Use proper code Format

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by AkanshaUniyal** [comment]
@ankitaggarwal23 sir, changes have been made.

File: Java/math/tribonacci.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+/*    Tribonacci Series:
+ 
+  Tribonacci series is a generalisatoin of Fibonacci series where the current element(greater than 0) is the sum of preceding three elements,
+  Let T be the Tribonacci series upto n terms then, T(n)=T(n-1)+T(n-2)+T(n-3).  */
+
+/*    Approach:
+
+   A term is the sum of preceding three terms in this series. The sum of three preceding variables will be stored in "sum" variable.
+   The first of the three variable terms will be overrided by second variable term which in turn will be overrided by third one and the third term will be 
+   overrided by the sum. This process will continue till n terms.  */
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Make it short and concise.

File: Java/math/tribonacci.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ public class tribonacci 
+ {
+    public static void main(String [] args)
+    { 
+
+        Scanner scan = new Scanner(System.in);
+        long num;
+	long first=0,second=0,third=1,sum=0,count=0;
+        System.out.println("Enter the number upto which the series is to be printed");
+        num=scan.nextLong();
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Align your code properly

File: Java/math/tribonacci.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+        while(num>0)	
+        {
+            count++;
+            if(count==1)	
+            System.out.print(0);
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Add one tab space

File: Java/math/tribonacci.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+            else if(count==2)
+            System.out.print(", "+0);
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Add one tab space

File: Java/math/tribonacci.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+            else if(count==3)
+            System.out.print(", "+1);
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Add one tab space

File: Java/math/tribonacci.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+            else    //Applicable to all the terms other than the first three terms of the series.
+            {
+            sum=first+second+third;	
+            System.out.print(", "+sum);
+            first=second;second=third;third=sum;  //Storing the values of succeeding terms in respective variables.
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Add one tab space and in last line ` first=second;second=third;third=sum;` give new line after semicolon

File: Java/math/tribonacci.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+ Enter the number upto which the series is to be printed
+ 23
+ 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81, 149, 274, 504, 927, 1705, 3136, 5768, 10609, 19513, 35890, 66012, 121415
+
+ Test Case 2:
+
+ Enter the number upto which the series is to be printed
+ 5
+ 0, 0, 1, 1, 2
+
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Remove extra test case one is enough

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by AkanshaUniyal** [comment]
@ankitaggarwal23 sir, changes have been made.

File: Java/math/tribonacci.java [hunk 12]
```diff
@@ -0,0 +1,64 @@
+ package mypack;
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Remove the package mypack;

File: Java/math/tribonacci.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+            {
+            	sum=first+second+third;	
+            		System.out.print(", "+sum);
+		    //Storing the values of succeeding terms in respective variables.
+            	first=second;
+		second=third;
+		third=sum;  
+            }
+             num--;    //Decreasing the value of total number of terms by one after each iteration.
+        }
```
Comment (ankitaggarwal23) [inline Java/math/tribonacci.java]: Align your code properly

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest looks good to merge.

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by ankitaggarwal23** [comment]
@AkanshaUniyal You have committed your code from master branch. Kindly don't commit from master branch that's why showing conflicts. Kindly open a new PR and commit from new branch not from main or master branch. Closing this PR as of now.

**Pull request closed without merging**